Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -106,27 +109,134 @@
 
 static const size_t maximumNumberOfShortcutCommands = 4;
 
-WebExtension::WebExtension(NSBundle *appExtensionBundle, NSURL *resourceBaseURL, RefPtr<API::Error>& outError)
+static String convertChromeExtensionToTemporaryZipFile(const String& inputFilePath)
+{
+    auto inputFileHandle = FileSystem::openFile(inputFilePath, FileSystem::FileOpenMode::Read);
+    if (!FileSystem::isHandleValid(inputFileHandle))
+        return nullString();
+
+    auto closeFile = makeScopeExit([&] {
+        FileSystem::unlockAndCloseFile(inputFileHandle);
+    });
+
+    static std::array<uint8_t, 4> expectedSignature = { 'C', 'r', '2', '4' };
+
+    // Verify Chrome extension magic signature.
```

Comment:
Super nit: This comment could be above the `expectedSignature` to indicate that those bits are the `Chrome extension magic signature`.